arch/risc-v/src/common/riscv_mtimer.c

```diff
@@ -104,7 +110,16 @@ static void riscv_mtimer_set_mtimecmp(struct riscv_mtimer_lowerhalf_s *priv,
                                       uint64_t value)
 {
 #ifdef CONFIG_ARCH_RV64
-  putreg64(value, priv->mtimecmp);
+  if (-1 == priv->mtime)
+    {
+      putreg32(0, priv->mtimecmp);
+      putreg32((uint32_t)(value >> 32), priv->mtimecmp + 4);
+      putreg32((uint32_t)value, priv->mtimecmp);
+    }
+  else
+    {
+      putreg64(value, priv->mtimecmp);
+    }
```

Comment:
It has been modified because 64bit/DWORD access to the cmp regs(low+high) at once (putreg64) is not allowed on this SoC, only WORD access sequence leads to the correct behavior of MTIMER.

Reply:
If so, please reorg your code to avoid the code dup:
```
#ifdef CONFIG_ARCH_RV64
  if (-1 != priv->mtime)
    {
      putreg64(value, priv->mtimecmp);
    }
  else
#endif
    {
      putreg32(UINT32_MAX, priv->mtimecmp + 4);
      putreg32(value, priv->mtimecmp);
      putreg32(value >> 32, priv->mtimecmp + 4);
    }
```

Reply:
This sequence is not the same as the current working code slice, we need to verify, then update if OK.

Reply:
Another thing, there may be a potential issue in this sequence: if the lower word of the "value" is 0xFFFFFFFF, an un-expected IRQ may raise before the mtimecmph gets updated, though it's rare.

Reply:
This sequence is suggested by spec(https://github.com/riscv/riscv-isa-manual/releases/download/Priv-v1.12/riscv-privileged-20211203.pdf):
```
# New comparand is in a1:a0.
li t0, -1
la t1, mtimecmp
sw t0, 0(t1) # No smaller than old value.
sw a1, 4(t1) # No smaller than new value.
sw a0, 0(t1) # New value.

Figure 3.29: Sample code for setting the 64-bit time comparand in RV32, assuming a little-endian
memory system and that the registers live in a strongly ordered I/O region. Storing -1 to the
low-order bits of mtimecmp prevents mtimecmp from temporarily becoming smaller than the lesser
of the old and new values.
```

Reply:
OK，got. Thanks！